src/builtins/Transformers.jl

```diff
@@ -341,27 +347,68 @@ names of features to be standardized.
     │ 3   │ 1.14708   │ 3     │
 
 """
-@with_kw_noshow mutable struct Standardizer <: Unsupervised
-    features::Vector{Symbol} = Symbol[] # features to be standardized; empty means all
+mutable struct Standardizer <: Unsupervised
+    # features to be standardized; empty means all
+    features::Union{AbstractVector{Symbol}, Function}
+    ignore::Bool # features to be ignored
+    ordered_factor::Bool
+    count::Bool
+end
+
+function MLJBase.clean!(transformer::Standardizer)
+    err = ""
+    if (
+        typeof(transformer.features) <: AbstractVector{Symbol} &&
+        isempty(transformer.features) &&
+        transformer.ignore
+    )
+        err *= "Features to be ignored must be specified in features field."
+    end
+    return err
+end
+
+# keyword constructor
+function Standardizer(
+    ;
+    features::Union{AbstractVector{Symbol}, Function}=Symbol[],
+    ignore::Bool=false,
+    ordered_factor::Bool=false,
+    count::Bool=false
+)
+    transformer = Standardizer(features, ignore, ordered_factor, count)
+    message = MLJBase.clean!(transformer)
+    isempty(message) || throw(ArgumentError(message))
+    return transformer
 end
 
 function MLJBase.fit(transformer::Standardizer, verbosity::Int, X)
     all_features = Tables.schema(X).names
-    mach_types   = collect(eltype(selectcols(X, c)) for c in all_features)
+    mach_types   = collect(elscitype(selectcols(X, c)) for c in all_features)
+    scitypes = [Continuous]
+    transformer.ordered_factor && push!(scitypes, OrderedFactor)
+    transformer.count && push!(scitypes, Count)
+    AllowedScitype = Union{scitypes...}
 
     # determine indices of all_features to be transformed
-    if isempty(transformer.features)
-        cols_to_fit = filter!(eachindex(all_features) |> collect) do j
-            mach_types[j] <: AbstractFloat
+    if typeof(transformer.features) <: AbstractVector{Symbol}
+        if isempty(transformer.features)
+            cols_to_fit = filter!(eachindex(all_features) |> collect) do j
+                mach_types[j] <: AllowedScitype
+            end
+        else
+            issubset(transformer.features, all_features) ||
+                @warn "Some specified features not present in table to be fit. "
+            cols_to_fit = filter!(eachindex(all_features) |> collect) do j
+                (all_features[j] in transformer.features && !transformer.ignore) &&
```

Comment:
@darrencl . I think this line always evaluates to `false` when transformer.ignore is `true`

Reply:
probably an `ifelse` function might be better here

Reply:
`ifelse(transformer.ignore, !(all_features[j] in transformer.features)  && mach_types[j] <: AllowedScitype,  (all_features[j] in transformer.features)  && mach_types[j] <: AllowedScitype )`

Reply:
You're right, I'll correct the test too. Thanks! :)